gomint-server/src/main/java/io/gomint/server/world/WorldAdapter.java

```diff
@@ -217,6 +245,9 @@ public Difficulty difficulty() {
      */
     @Override
     public Collection<EntityPlayer> onlinePlayers() {
+        if (!mainThread()) {
+            this.logger.warn("Getting players not from world's thread. This is not safe", new Exception());
```

Comment:
This does spam GoMint.onlinePlayers()? Should the API be changed?

Reply:
The players map is not thread safe and the method only returns an unmodifiable view and does not copy.

Reply:
Problem is that only the view itself is unmodifiable. This means that when `this.players` changes it still will CME whilst onlinePlayers() iterates. I guess a defensive copy would be best here since it would also get rid of the problem when players are in two worlds (which they could when they teleport whilst querying for GoMint#onlinePlayers())

Reply:
Another possible option would be to maintain another serverwide set of players just for this api.
Defensive copy could also CME.

Reply:
another idea would be to use a stream or consumer and change that API all together

Reply:
> another idea would be to use a stream or consumer and change that API all together

You mean that it schedules the player consumer on every world thread sync?

Reply:
yes

Reply:
What about making `GoMint.onlinePlayers()` just return `this.playersByUUID.values()`?

I wil still add the player consumer on every world thread, that sounds potentially useful for plugins to me.

Reply:
Thats ok, could we reduce the API a bit by making the Consumer based method a overload from the `onlinePlayers()`? So we end up with `onlinPlayers(Consumer)`